src/Sylius/Component/Core/Calculator/ProductVariantPriceCalculator.php

```diff
@@ -30,9 +30,10 @@ public function calculate(ProductVariantInterface $productVariant, array $contex
 
         if (null === $channelPricing) {
             throw new MissingChannelConfigurationException(sprintf(
-                'Channel %s has no price defined for product variant %s',
+                'Channel %s has no price defined for product variant %s (%s)',
                 $context['channel']->getName(),
-                $productVariant->getName()
+                $productVariant->getName(),
+                $productVariant->getCode()
```

Comment:
Imo, it would still look quite confusing if there is no name - `Channel DEFAULT has no price defined for product variant  (VARIANT_CODE)` (double space 💃). What about fallback to code if there is no name? `$productVariant->getName() ?? $productVariant->getCode()`?

Then we have either

`Channel DEFAULT has no price defined for product variant Variant name`

or

`Channel DEFAULT has no price defined for product variant VARIANT_CODE`

Reply:
It's a good idea, but the code is the most valuable. Much easier to find, both in a db viewer or in the admin interface. I will make it nicer :)

Reply:
What do you think now?